@types/superagent error TS2304: Cannot find name 'XMLHttpRequest'.

- [ ] I tried using the latest `xxxx/xxxx.d.ts` file in this repo and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] I want to talk about `xxxx/xxxx.d.ts`.
  - The authors of that type definition are cc/ @....

Today my builds start failing due to errors on @type/superagent. I start lowering the version number until i found that the problem start with version 2.0.34. Before that no error is produced by the Typescript compiler (using Version 2.1.0-dev.20161017)

With @types/superagent@2.0.34 the error is:
  node_modules/@types/superagent/index.d.ts(79,12): error TS2304: Cannot find name 'XMLHttpRequest'.

And with @types/superagent@2.0.35 the error is:
  node_modules/@types/superagent/index.d.ts(79,12): error TS2304: Cannot find name 'XMLHttpRequest'.

I hope this info help you guys.

You can add a superagent.d.ts file with these contents:

```

// error TS2304: Cannot find name 'XMLHttpRequest'
declare interface XMLHttpRequest {}
// error TS2304: Cannot find name 'Blob'
declare interface Blob {}
```

> // error TS2304: Cannot find name 'XMLHttpRequest'
> declare interface XMLHttpRequest {}
> // error TS2304: Cannot find name 'Blob'
> declare interface Blob {}

@JasonKleban where does this file go? In `node_modules` > `superagent`? I've been trying to figure this out and I'm at my wits' end.

--- a/types/promisify-supertest/index.d.ts
+++ b/types/promisify-supertest/index.d.ts
@@ -2,7 +2,7 @@
 // Project: https://www.npmjs.com/package/promisify-supertest
 // Definitions by: Leo Liang <https://github.com/aleung>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.0
 
 // Mostly copy-pasted from supertest.d.ts
 

--- a/types/simple-cw-node/index.d.ts
+++ b/types/simple-cw-node/index.d.ts
@@ -2,7 +2,7 @@
 // Project: https://github.com/astronaughts/simple-cw-node
 // Definitions by: vvakame <https://github.com/vvakame>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.0
 
 import superagent = require("superagent");
 // TODO 1. update superagent with generics

--- a/types/superagent-bunyan/index.d.ts
+++ b/types/superagent-bunyan/index.d.ts
@@ -2,7 +2,7 @@
 // Project: https://github.com/joaquimserafim/superagent-bunyan#readme
 // Definitions by: Alex Brick <https://github.com/bricka>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.0
 
 import * as Logger from 'bunyan';
 import { Plugin } from 'superagent';

--- a/types/superagent-no-cache/index.d.ts
+++ b/types/superagent-no-cache/index.d.ts
@@ -2,7 +2,7 @@
 // Project: https://github.com/johntron/superagent-no-cache
 // Definitions by: Michael Ledin <https://github.com/mxl>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.0
 
 import * as request from 'superagent';
 

--- a/types/superagent-prefix/index.d.ts
+++ b/types/superagent-prefix/index.d.ts
@@ -2,7 +2,7 @@
 // Project: https://github.com/johntron/superagent-prefix, https://github.com/mattahorton/superagent-prefix
 // Definitions by: Michael Ledin <https://github.com/mxl>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.0
 
 import * as request from 'superagent';
 

--- a/types/superagent/index.d.ts
+++ b/types/superagent/index.d.ts
@@ -8,10 +8,12 @@
 //                 Adam Haglund <https://github.com/beeequeue>
 //                 Lukas Elmer <https://github.com/lukaselmer>
 //                 Jesse Rogers <https://github.com/theQuazz>
+//                 Chris Arnesen <https://github.com/carnesen>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.0
 
 /// <reference types="node" />
+/// <reference lib="dom" />
 
 import * as fs from 'fs';
 import * as https from 'https';

--- a/types/supertest/index.d.ts
+++ b/types/supertest/index.d.ts
@@ -3,7 +3,7 @@
 // Definitions by: Alex Varju <https://github.com/varju>
 //                 Petteri Parkkila <https://github.com/pietu>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 3.0
 
 import * as superagent from "superagent"
 

